### PR TITLE
feat(cloudbuild): generate library google/cloud/cloudbuild

### PR DESCRIPTION
**Pull request by devjgm**
Fixes: #7892

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8019)

### CODE REVIEW

File: google/cloud/cloudbuild/cloud_build_connection.cc [hunk 1]
```diff
@@ -0,0 +1,679 @@
...
+          return stub->AsyncGetOperation(cq, std::move(context), request);
+        },
+        [stub](google::cloud::CompletionQueue& cq,
+               std::unique_ptr<grpc::ClientContext> context,
+               google::longrunning::CancelOperationRequest const& request) {
+          return stub->AsyncCancelOperation(cq, std::move(context), request);
+        },
+        &google::cloud::internal::ExtractLongRunningResultResponse<
+            google::devtools::cloudbuild::v1::Build>,
+        retry_policy_prototype_->clone(), backoff_policy_prototype_->clone(),
```
Comment (dbolduc) [inline google/cloud/cloudbuild/cloud_build_connection.cc:232]: Please rebase and regenerate to pick up the generator changes from #8013

Comment (devjgm) [reply google/cloud/cloudbuild/cloud_build_connection.cc]: Done.

**Review by dbolduc — APPROVED** [review]

File: google/cloud/cloudbuild/cloud_build_connection.cc (hunk 1 above)
Comment (dbolduc) [reply google/cloud/cloudbuild/cloud_build_connection.cc]: Thanks

File: google/cloud/cloudbuild/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+* Bazel and CMake builds
+
+## Documentation
+
+* Official documentation about the [Cloud Build API][cloud-service-docs] service
+* [Reference doxygen documentation][doxygen-link] for each release of this
+  client library
+* Detailed header comments in our [public `.h`][source-link] files
+
+[cloud-service-docs]: https://cloud.google.com/cloudbuild
```
[the remark this answers is not in the document]
Comment (devjgm) [reply google/cloud/cloudbuild/README.md]: Fixed.

File: google/cloud/cloudbuild/doc/main.dox [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+For operations that do not return a value the library simply returns
+`google::cloud::Status`.
+
+### Retry, Backoff, and Idempotency Policies.
+
+The library automatically retries requests that fail with transient errors, and
+uses [exponential backoff] to backoff between retries. Application developers
+can override the default policies.
+
+[cloud-service-docs]: https://cloud.google.com/cloudbuild
```
[the remark this answers is not in the document]
Comment (devjgm) [reply google/cloud/cloudbuild/doc/main.dox]: Fixed

**Review by coryan — APPROVED** [review]

**Pull request merged**